openeogeotrellis/load_stac.py

```diff
@@ -458,7 +469,7 @@ def intersects_temporally(interval) -> bool:
                        .withNominalDate(itm.properties.get("datetime") or itm.properties["start_datetime"]))
 
             for asset_id, asset in band_assets.items():
-                asset_band_names = get_band_names(itm, asset) or [asset_id]
+                asset_band_names = get_band_names(item=itm, asset=asset) or [asset_id]
```

Comment:
here we may need to check these band names inside an asset for overlap with other assets.
We now know that it is possible that two assets share a band name, so we need to resolve this before sending assets into geotrellis for datacube building. This seems like the best location.

Proposal: sort by gsd from smallest to larger, fallback to alphabetic order to make it deterministic?
Then keep a set of band names that have already been seen, and only add links for new names. Make sure to always include asset_id in list of band names.

Reply:
I tried to implement this proposal in https://github.com/Open-EO/openeo-geopyspark-driver/pull/1213/commits/2d8422af2bc78001e02ca1467cd581b44ba1a3fc,
which reverts the change to the `test_lcfm_improvements` test